### PR TITLE
feat(checkbox): update checkbox

### PR DESCRIPTION
**Pull request by thisisandy**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
	As a developer, I want my checkbox compatible with Vue 3.0
> 2. Resolve what problem.
	Updated conflict api between existing checkbox implementation with Vue 3.0 RFCs
> 3. Related issue link.
	#1913 

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

**Comment by thisisandy** [comment]
@Amour1688 Please help review

File: components/checkbox/Checkbox.jsx [hunk 1]
```diff
@@ -74,41 +78,40 @@ export default {
   },
 
   render() {
-    const { checkboxGroupContext: checkboxGroup, $slots } = this;
     const props = getOptionProps(this);
-    const children = $slots.default;
-    const { mouseenter = noop, mouseleave = noop, input, ...restListeners } = getListeners(this);
-    const { prefixCls: customizePrefixCls, indeterminate, ...restProps } = props;
+    const { checkboxGroupContext: checkboxGroup, $slots, $attrs } = this;
+    const children = $slots.default();
```
Comment (Amour1688) [inline components/checkbox/Checkbox.jsx:83]: $slots.default && $slots.default()

File: components/checkbox/index.js [hunk 2]
```diff
@@ -1,12 +1,10 @@
 import Checkbox from './Checkbox';
 import CheckboxGroup from './Group';
-import Base from '../base';
 
 Checkbox.Group = CheckboxGroup;
 
 /* istanbul ignore next */
 Checkbox.install = function(Vue) {
-  Vue.use(Base);
   Vue.component(Checkbox.name, Checkbox);
```
Comment (Amour1688) [inline components/checkbox/index.js:8]: app.component

File: components/checkbox/Checkbox.jsx [hunk 3]
```diff
@@ -74,41 +78,40 @@ export default {
...
-    const children = $slots.default;
-    const { mouseenter = noop, mouseleave = noop, input, ...restListeners } = getListeners(this);
-    const { prefixCls: customizePrefixCls, indeterminate, ...restProps } = props;
+    const { checkboxGroupContext: checkboxGroup, $slots, $attrs } = this;
+    const children = $slots.default && $slots.default();
+    const { indeterminate, prefixCls: customizePrefixCls, ...restProps } = props;
     const getPrefixCls = this.configProvider.getPrefixCls;
     const prefixCls = getPrefixCls('checkbox', customizePrefixCls);
-
+    const { onMouseenter, onMouseleave, onInput, ...restAttrs } = $attrs;
```
Comment (tangjinzhou) [inline components/checkbox/Checkbox.jsx:87]: class style 也要拿出来，并添加到 label 上   vue3 中inheritAttrs也会影响 class style 的挂载

File: components/checkbox/Checkbox.jsx [hunk 4]
```diff
@@ -74,41 +78,40 @@ export default {
   },
 
   render() {
```
Comment (tangjinzhou) [inline components/checkbox/Checkbox.jsx:80]: 删除 model 字段
this.$emit('input', targetChecked);
to
this.$emit('update:checked', targetChecked);

File: components/checkbox/Group.jsx [hunk 5]
```diff
@@ -16,14 +17,18 @@ export default {
...
     return {
       checkboxGroupContext: this,
     };
   },
-  inject: {
-    configProvider: { default: () => ConfigConsumerProps },
+  setup() {
+    return {
+      configProvider: inject('configProvider', ConfigConsumerProps),
+      checkboxGroupContext: provide('checkboxGroupContext', this),
```
Comment (tangjinzhou) [inline components/checkbox/Group.jsx:30]: setup里拿不到this
provide 可以放在 created周期里

File: components/checkbox/Group.jsx [hunk 6]
```diff
@@ -16,14 +17,18 @@ export default {
     value: PropTypes.array,
     options: PropTypes.array.def([]),
     disabled: PropTypes.bool,
+    onChange: PropTypes.func,
   },
   provide() {
```
Comment (tangjinzhou) [inline components/checkbox/Group.jsx:22]: 删除

File: components/checkbox/Group.jsx [hunk 7]
```diff
@@ -16,14 +17,18 @@ export default {
     value: PropTypes.array,
     options: PropTypes.array.def([]),
     disabled: PropTypes.bool,
+    onChange: PropTypes.func,
```
Comment (tangjinzhou) [inline components/checkbox/Group.jsx:20]: why？

**Comment by tangjinzhou** [comment]
1、lint
2、单测

**Pull request merged**